runconfig/hostconfig.go

```diff
@@ -96,38 +109,46 @@ func (n PidMode) Valid() bool {
 	return true
 }
 
+// DeviceMapping represents the device mapping between the host and the container
 type DeviceMapping struct {
 	PathOnHost        string
 	PathInContainer   string
 	CgroupPermissions string
 }
 
+// RestartPolicy represents the restart policies of the container
 type RestartPolicy struct {
 	Name              string
 	MaximumRetryCount int
 }
 
+// IsNone indicates whether the container has no restart policy
 func (rp *RestartPolicy) IsNone() bool {
 	return rp.Name == "no"
 }
 
+// IsAlways indicates whether the container has the always restart policy
 func (rp *RestartPolicy) IsAlways() bool {
 	return rp.Name == "always"
 }
 
+// IsOnFailure indicates whether the container has the on-failure restart policy
```

Comment:
quote "on-failure" just to be consistent with the previous func - even though in this case its not as needed.